src/main/scala/org/broadinstitute/dsde/workbench/avram/db/DbReference.scala

```diff
@@ -0,0 +1,45 @@
+package org.broadinstitute.dsde.workbench.avram.db
+
+import org.apache.commons.dbcp2.BasicDataSource
+import org.broadinstitute.dsde.workbench.avram.config.DbcpDataSourceConfig
+import slick.dbio.DBIO
+import slick.jdbc.{JdbcProfile, TransactionIsolation}
+import scala.concurrent.{ExecutionContext, Future}
+
+import AvramPostgresProfile.api._
+
+case class DbReference(private val dataSourceConfig: DbcpDataSourceConfig)(implicit val executionContext: ExecutionContext) {
+
+  val dataAccess = new DataAccess(AvramPostgresProfile)
+
+  val ds = new BasicDataSource()
+  ds.setDriverClassName(dataSourceConfig.driverClassName)
+  ds.setUrl(dataSourceConfig.url)
+  ds.setUsername(dataSourceConfig.username)
+  ds.setPassword(dataSourceConfig.password)
+  ds.setMaxTotal(dataSourceConfig.maxTotal)
+
+  val database = Database.forDataSource(ds, Option(ds.getMaxTotal), AsyncExecutor("Avram Executor", dataSourceConfig.slickNumThreads, dataSourceConfig.slickQueueSize))
+
+  def inTransaction[T](f: (DataAccess) => DBIO[T], isolationLevel: TransactionIsolation = TransactionIsolation.RepeatableRead): Future[T] = {
```

Comment:
I think ReadCommitted is the default postgres isolation level. Maybe it should be the default here.

(Should think more about what type of transactions avram needs -- not sure at the moment).

Reply:
Yes, `ReadCommitted` is the default for postgres. I think we should stick with that if we can. It's a little extra work to consider the consequences in application code, but I think it's worth it to help limit transaction/lock contention.